controller/volume_controller.go

```diff
@@ -549,9 +549,12 @@ func (vc *VolumeController) ReconcileEngineReplicaState(v *longhorn.Volume, es m
 	if e.Status.CurrentState != longhorn.InstanceStateRunning {
 		// If a replica failed at attaching stage before engine become running,
 		// there is no record in e.Status.ReplicaModeMap
+		engineInstanceCreationCondition := types.GetCondition(e.Status.Conditions, longhorn.InstanceConditionTypeInstanceCreation)
+		isNoAvailableBackend := strings.Contains(engineInstanceCreationCondition.Message, fmt.Sprintf("exit status %v", int(syscall.ENODATA)))
```

Comment:
Previously I tried to launch an error struct so that Longhorn can easily marshal and unmarshal error codes (even among different components). I am not sure if we can popularize it and make it a separate repo. The idea is from the gRPC error handling. WDYT?

Reply:
Sounds good. Let's work the idea in https://github.com/longhorn/longhorn/issues/4996. WDYT?